src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/AvoidUninstantiatedInternalClasses.cs

```diff
@@ -101,6 +103,17 @@ public override void Initialize(AnalysisContext analysisContext)
                     {
                         instantiatedTypes.TryAdd(type.BaseType, null);
                     }
+
+                    // Consider class type declared in the CoClass attribute as instantiated
+                    if (type.TypeKind == TypeKind.Interface &&
+                        type.GetAttributes().FirstOrDefault(x => x.AttributeClass.Equals(coClassAttributeSymbol)) is AttributeData coClassAttribute &&
```

Comment:
I am not checking whether the interface or class do have the `Guid` attribute. I am not sure whether this rule should be strict on having the coclass properly defined.

Reply:
Yes, I like your current implementation.